src/components/Map/ChallengeMap.tsx

```diff
@@ -1,105 +1,102 @@
+/* module imports */
 import DeckGL from '@deck.gl/react';
+import React, {
+  useCallback,
+  useState,
+} from 'react';
+import { InteractiveMap } from 'react-map-gl';
 import { PathLayer } from '@deck.gl/layers';
-// import d3 from 'd3-ease'; // 3rd-party easing functions
 import { ViewState } from 'react-map-gl/src/mapbox/mapbox';
 
-// local imports
-// import flatpin from './assets/images/red-pin.png';
-import { Track } from '../../model/ChallengeConfiguration';
+/* local imports */
+import {
+  Challenge,
+  Track,
+} from '../../model/ChallengeConfiguration';
+import { Circuit } from '../Circuit/Circuit';
 
+/* interfaces & types */
 interface PropTypes {
   tracks: Track[];
 }
 
-const ChallengeMap = ({ tracks }: PropTypes) => {
-  const defaultView: ViewState = {
-    latitude: 37.78,
-    longitude: -122.45,
-    zoom: 12,
-  };
+/* helpers & constants */
+// This will initialize a challenge from a couple of lines.
+const defaultChallenge: Challenge = {
+  start: {
+    firstPoint: {
+      lon: -122.4,
+      lat: 37.7,
+    },
+    secondPoint: {
+      lon: -122.4,
+      lat: 37.8,
+    },
+  },
+  finish: {
+    firstPoint: {
+      lon: -122.5,
+      lat: 37.7,
+    },
+    secondPoint: {
+      lon: -122.5,
+      lat: 37.8,
+    },
+  },
+};
 
-  const [viewState, setViewState] = React.useState<ViewState>(defaultView);
+// This constant defaults to showing San Francisco.
+const defaultView: ViewState = {
+  latitude: 37.78,
+  longitude: -122.45,
+  zoom: 12,
+};
 
-  const handleViewStateChange = React.useCallback(
-    (args: { viewState: any }) => {
+/* react component */
+const ChallengeMap = ({ tracks }: PropTypes) => {
+  // Map's viewstate
+  const [viewState, setViewState] = useState<ViewState>(defaultView);
+  const handleViewStateChange = useCallback(
+    (args: { viewState: ViewState }) => {
       setViewState(args.viewState);
     },
     [],
   );
 
-  const goToSF = React.useCallback(() => {
-    setViewState((viewport: ViewState) => {
-      // When deck-gl is initialized the viewport contains additional keys width and height
-      // which are required to create a WebMercatorViewport and calculate bounds.
-      if ('width' in viewport && 'height' in viewport) {
-        const { longitude, latitude, zoom } = new WebMercatorViewport(
-          viewport as ViewState & { width: number; height: number },
-        ).fitBounds(
-          [
-            [-122.4, 37.7],
-            [-122.5, 37.8],
-          ],
-          {
-            padding: 20,
-            offset: [0, -100],
-          },
-        );
-
-        return {
-          ...viewport,
-          longitude,
-          latitude,
-          zoom,
-          transitionDuration: 5000,
-          transitionInterpolator: new FlyToInterpolator(),
-          // transitionEasing: d3.easeCubic
-        };
-      }
-
-      return viewport;
-    });
-  }, []);
+  // Challenge's start & finish states
+  const challenge = (defaultChallenge);
```

Comment:
I think the challenge configuration should probably be a prop to ChallengeMap

Reply:
Using:
```js
/* react component */
const ChallengeMap = ({
  tracks,
  challenge=defaultChallenge,
}: PropTypes) => {
```
until defaultChallenge is replaced